### PR TITLE
Fix a bunch of errors in `color_brewer` and add unit tests

### PR DESCRIPTION
**Pull request by ajabep**
The non-qualitative schemes should be extendable. This commit adds this feature.

Fixes https://github.com/python-visualization/branca/issues/104 and https://github.com/python-visualization/branca/issues/114. Also probably fix https://github.com/python-visualization/folium/issues/1270.

Moreover, the following error was not reported via issues and are fixed:
* TypeError in the linear_gradient function when trying to extend any scheme. Indeed, in color_brewer, the key searched in the scheme database was not found, thus, it was passing `None` instead of a real scheme vector to linear_gradient;
* The scheme 'viridis' was not in the base_codes JSON;
* Multiple scheme hadn't any metadata in scheme_info JSON;
* When a `n` value provided to `color_scheme` was a float, it tried to select an unknown scheme without raising the right Exception type.
* The unit tests was using the deprecated (and remove) method `WebDriver.find_element_by_css_selector`.

### CODE REVIEW

**Review by ocefpaf — APPROVED** [review]
Awesome! LGTM but I've been away from folium+branca for so long that I'd rather wait for @Conengmo to take a second look before merging.

File: branca/utilities.py [hunk 1]
```diff
@@ -162,10 +163,20 @@ def color_brewer(color_code, n=6):
...
         else:
+            longest_scheme_name = base_code
+            longest_scheme_n = 0
+            for sn_name in schemes.keys():
+                if '_' not in sn_name:
+                    continue
+                if sn_name.split('_')[0] != base_code:
+                    continue
+                if int(sn_name.split('_')[1]) > longest_scheme_n:
+                    longest_scheme_name = sn_name
```
Comment (Conengmo) [inline branca/utilities.py:174]: shouldn't we update `longest_scheme_n` here as well?

**Review by Conengmo — CHANGES_REQUESTED** [review]
Thanks so much for your PR. I'm happy you fixed these glaring issues with the color brewer and also the added tests!

I have one comment regarding the longest scheme name, could you take a look please?

File: branca/utilities.py (hunk 1 above)
Comment (ajabep) [reply branca/utilities.py:174]: Nice catch!
It's fixed

**Comment by Conengmo** [comment]
Thanks again @ajabep!

**Pull request merged**